ecl/ClassicTestModified.ecl

```diff
@@ -0,0 +1,189 @@
+          '''layers.Dense(1, activation=None)'''];
+
+// compileDef defines the compile line to use for compiling the defined model.
+// Note that 'model.' is implied, and should not be included in the compile line.
+compileDef := '''compile(optimizer=tf.keras.optimizers.SGD(.02),
+              loss=tf.keras.losses.mean_squared_error,
+              metrics=[tf.keras.metrics.mean_squared_error])
+              ''';
+
+// Note that the order of the GNNI functions is maintained by passing tokens returned from
+// one call into the next call that is dependent on it.
+// For example, s is returned from GetSession().  It is used as the input to
+// DefineModels(...) so
+// that DefineModels() cannot execute until GetSession() has completed.
+// Likewise, mod, the output from GetSession() is provided as input to Fit().  Fit in turn
+// returns a token that is used by GetLoss(), EvaluateMod(), and Predict(),
+// which are only dependent on Fit() having completed, and are not order
+// dependent on one another.
+
+// GetSession must be called before any other functions
+Session := GNNI.GetSession();
+
+// DefineModel is dependent on the Session
+//   ldef contains the Python definition for each Keras layer
+//   compileDef contains the Keras compile statement.
+Model := GNNI.DefineModel(Session, ldef, compileDef);
+
+// Fit trains the models, given training X and Y data.  BatchSize is not the Keras batchSize,
+// but defines how many records are processed on each node before synchronizing the weights
+Model2 := GNNI.Fit(Model, TrainIndTensor, TrainDepTensor, batchSize := batchSize, numEpochs := numEpochs,
+                      trainToLoss := trainToLoss, learningRateReduction := lrr,
+                      batchSizeReduction := bsr);
+
+// Predict computes the neural network output given a set of inputs.
+Prediction := GNNI.Predict(Model2, TestIndTensor);
+
+// Note that the Tensor is a packed structure of Tensor slices.  GetData()
+// extracts the data into a sparse cell-based form -- each record represents
+// one Tensor cell.  See Tensor.R4.TensData.
+TestDepData := Tensor.R4.GetData(TestDepTensor);
+PredData := Tensor.R4.GetData(Prediction);
+
+// Here  we are comparing the expected values (testYDat) with the predicted values (predDat)
+// and calculating the error and squared error to validate the process
+cmp := JOIN(TestDepData, PredData, LEFT.indexes[1] = RIGHT.indexes[1], TRANSFORM({SET OF REAL indexes,
+                                              REAL pred, REAL actual, REAL error, REAL sqError},
+                  SELF.indexes := LEFT.indexes, SELF.pred := RIGHT.value, SELF.actual := LEFT.value,
+                  SELF.error := ABS(SELF.actual - SELF.pred),
+                  SELF.sqError := SELF.error * SELF.error), LEFT OUTER, LOCAL);
+
+Result := RECORD
+  REAL err := cmp.error;
+END;
+
+Errors := TABLE(cmp, Result);
+
+// No errors should be above 0.1
+HighErrors := COUNT(Errors(err > 0.1));
```

Comment:
That's what we are talked yesterday. On my Ubuntu 18.04 + Tensorflow 2.6.2 this check (> 0.1) results > 200 errors.
If I change it to 0.2 then I can get only a couple of errors, and with 0.3 it is passed,
So, the question is 0.3 limit is acceptable for Tensor flow 2.9.1 or not?
If the answer is yes, we can use 0.3 for that check. 
If not, then I should ensure it is always executed on >=2.9.1. 
Or we can use stored parameter to input Tensor flow version (the automated test system can query it and pass forward) and set the limit value based on the stored parameter.